app/models/message.rb

```diff
@@ -325,4 +329,39 @@ def set_conversation_activity
     conversation.update_columns(last_activity_at: created_at)
     # rubocop:enable Rails/SkipsModelValidations
   end
+
+  def encrypt_msg
+    key = ENV.fetch('ENCRYPTION_KEY', '')
+    if key
+      cipher = OpenSSL::Cipher.new('aes-256-cbc')
+      cipher.encrypt
+      cipher.key = key
+      iv = cipher.random_iv
+      data = { :message => content }
+
+      encrypted = cipher.update(data.to_json) + cipher.final
+      self.content = Base64.encode64(iv + encrypted)
+    end
+  end
+
+  def decrypt_msg
+    key = ENV.fetch('ENCRYPTION_KEY', '')
+    if key && base64?(content)
+      decipher = OpenSSL::Cipher.new('aes-256-cbc')
+      decipher.decrypt
+      decoded_str = Base64.decode64(content)
+      decipher.key = key
+      decipher.iv = decoded_str[0...16]
+      plain_string = decipher.update(decoded_str[16...]) + decipher.final
+
+      if plain_string
+        plain = JSON.parse(plain_string)
+        self.content = plain['message']
+      end
+    end
+  end
+
+  def base64?(value)
```

Comment:
I think we can skip this. this will unnecessarily add base64 encode and decode cost. I think we won't need this on any existing system. even if we need to implement this on any existing environment then we can write the migration script.

Reply:
Actually this is needed as we are using at initialize event. so, if we're creating new message instance it gives error